src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -297,6 +293,27 @@ export function getMostRecentBundleBlockRanges(
   return bundleData;
 }
 
+/**
+ * Resolves the corresponding deposit for a fill. Unlike in the Pre UBA clients, this function does NOT
+ * fall back to querying fresh RPC events to try to find a fill. Instead, if the deposit can't be found
+ * then this code will just crash, protecting the caller's funds. This is because if we were to find an old
+ * deposit, we'd need to recompute its expected realizedLpFeePct, which would be based on the deposit balancing
+ * fee and therefore requires more information from the flows preceding it. This is a future TODO.
+ * @param fill The fill to resolve the corresponding deposit for
+ * @param spokePoolClients The spoke clients to query for the deposit
+ * @returns The corresponding deposit for the fill, or undefined if the deposit was not found
+ */
+export async function resolveCorrespondingDepositForFill(
+  fill: FillWithBlock,
+  spokePoolClients: SpokePoolClients,
+  hubPoolClient: HubPoolClient
+): Promise<DepositWithBlock | undefined> {
+  // Matched deposit for fill was not found in spoke client. This situation should be rare so let's
+  // send some extra RPC requests to blocks older than the spoke client's initial event search config
+  // to find the deposit if it exists.
+  return queryHistoricalDepositForFill(hubPoolClient, spokePoolClients, fill);
+}
```

Comment:
Is this currently just a wrapper over `resolveCorrespondingDepositForFill`

Reply:
yeah basically